lib/spack/spack/test/container/test_docker.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import spack.container.writers as writers
+
+
+def test_manifest(minimal_configuration):
+    writer = writers.create(minimal_configuration)
+    manifest_str = writer.manifest
+    for line in manifest_str.split('\n'):
+        assert 'echo' in line
```

Comment:
This seems like a weak test. What are we trying to ensure here?

Reply:
That we're calling without errors the overridden `DockerContext.manifest` property:

https://github.com/spack/spack/blob/b2d1a6da51a347110190f9ddc5ec9e7317331001/lib/spack/spack/container/writers/docker.py#L17-L30

which adds this `echo` to each line since we can't use heredoc notation in a `Dockerfile`.